File: Substrings_of_Size_Three_with_Distinct_Characters/Substrings_of_Size_Three_with_Distinct_Characters.py
s = "xyzzaz"
def countGoodSubstrings(s):
    count=0
    for i in range(len(s)):
        str1=""
        for j in range(i,len(s)):
            str1+=s[j]
            s1=list(set(str1))
            if len(s1)== 3 and len(str1) == 3:
                count+=1

    return count

print(countGoodSubstrings(s))

File: Substrings_of_Size_Three_with_Distinct_Characters/Substrings_of_Size_Three_with_Distinct_Characters.go
package main

import (
	"fmt"
)

func main() {
	s := "xyzzaz"
	// s := "aababcabc"
	fmt.Println(countGoodSubstrings(s))
}

func countGoodSubstrings(s string) int {
	count := 0
	for i := 0; i < len(s); i++ {
		str1 := ""
		mk := make(map[string]int)

		for j := i; j < len(s); j++ {
			str1 += string(s[j])
			mk[string(s[j])] = count
			if len(str1) == 3 && len(mk) == 3 {
				count += 1
			}
		}
	}
	return count
}
